bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.salus.internal.handler;
+
+import static java.util.Collections.emptySortedSet;
+import static java.util.Objects.requireNonNullElse;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.openhab.core.thing.ThingStatus.OFFLINE;
+import static org.openhab.core.thing.ThingStatus.ONLINE;
+import static org.openhab.core.thing.ThingStatusDetail.CONFIGURATION_ERROR;
+import static org.openhab.core.types.RefreshType.REFRESH;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.concurrent.ScheduledFuture;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.salus.internal.rest.Device;
+import org.openhab.binding.salus.internal.rest.DeviceProperty;
+import org.openhab.binding.salus.internal.rest.GsonMapper;
+import org.openhab.binding.salus.internal.rest.JettyHttpClient;
+import org.openhab.binding.salus.internal.rest.SalusApi;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public final class CloudBridgeHandler extends BaseBridgeHandler implements CloudApi {
+    private Logger logger = LoggerFactory.getLogger(CloudBridgeHandler.class.getName());
+    private final HttpClientFactory httpClientFactory;
+    @NonNullByDefault({})
+    private LoadingCache<String, SortedSet<DeviceProperty<?>>> devicePropertiesCache;
+    @Nullable
+    private SalusApi salusApi;
+    @Nullable
+    private ScheduledFuture<?> scheduledFuture;
+
+    public CloudBridgeHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            internalInitialize();
+        } catch (Exception ex) {
```

Comment:
What would happen if `internalInitialize` will throw any `(Runtime)Exception`? Is it handled by OH? If yes then how?

To give you an example of some unexpected error: `scheduledPool.scheduleWithFixedDelay` this can throw `RejectedExecutionException` that is nowhere handled. Other methods also can throw exceptions (that might not even be enumerated in java docs)

Reply:
`scheduleWithFixedDelay` is used in many bindings, i have not seen a case that has a try/catch. Exception handling should be as specific as possible and handled as close as possible to the place it was thrown.

Reply:
`scheduleWithFixedDelay` it's just an example[1] so please do not focus on this case: the true question that I would like to know answer to is: 

 > What would happen if `internalInitialize` will throw any `(Runtime)Exception`? Is it handled by OH? If yes then how?

[1] Regarding `scheduleWithFixedDelay` :

If you will check the Java doc of this method you can see that this method might throw 3 exceptions (in our case only one of them can happen). If other binding are not dealing with `RejectedExecutionException` then who is?

```
    /**
     * Submits a periodic action that becomes enabled first after the
     * given initial delay, and subsequently with the given delay
     * between the termination of one execution and the commencement of
     * the next.
     * [...]
     * @throws RejectedExecutionException if the task cannot be
     *         scheduled for execution
     * @throws NullPointerException if command or unit is null
     * @throws IllegalArgumentException if delay less than or equal to zero
     */
```

Reply:
I don;t see any binding catching that exception, so im sure it wont be thrown. Please remove this block.

Reply:
So it will be handled? Any exception?

Reply:
There is no higher level Exception handling in the framework that i know of. Checked Exceptions should be used anyway, see other comment regarding Exceptions.

Reply:
done